### PR TITLE
Port react jsx files to typescript

### PR DESCRIPTION
**Pull request by jonasem**
Rewrite the remaining files (jsx) to typescript. After this PR all the code, except tests will be typed.

### CODE REVIEW

File: src/About/DeviceCard.tsx [hunk 1]
```diff
@@ -49,12 +49,13 @@ export default () => {
             <Section title="ID">{device.serialNumber}</Section>
             <Section title="PCA">{pca || 'Unknown'}</Section>
             <Section title="Cores">{cores || 'Unknown'}</Section>
+            {/*
```
Comment (jonasem) [inline src/About/DeviceCard.tsx:52]: This is an existing bug, doesnt work today. Commented out and made a trello https://trello.com/c/9XBs3jZF/169-fix-device-info-in-about-pane

File: src/Device/DeviceSetup/DeviceSetupView.tsx [hunk 2]
```diff
@@ -37,13 +49,17 @@ export default class DeviceSetupDialog extends React.Component {
             this.props;
         const { selectedChoice } = this.state;
 
-        if (choices && (choices.length > 0 || choices.size > 0)) {
+        if (
+            choices &&
+            (choices.length > 0 ||
+                (choices as unknown as { size: number }).size > 0)
+        ) {
```
Comment (nicolai-nordic) [inline src/Device/DeviceSetup/DeviceSetupView.tsx:56]: Can `choices` be either a list of strings or an object with a `size` property?

Comment (nicolai-nordic) [reply src/Device/DeviceSetup/DeviceSetupView.tsx:56]: Is this some remnants from `Immutable.js` or something that we can just remove?

**Review by nicolai-nordic — APPROVED** [review]
Very long PR, so it's difficult to get a good overview, but it looks good from what I can see.

File: src/Device/DeviceSetup/DeviceSetupView.tsx (hunk 2 above)
Comment (jonasem) [reply src/Device/DeviceSetup/DeviceSetupView.tsx:56]: Good catch! It's actually not exported, so no other apps are using it directly, and internally in shared we are only using string.

**Comment by datenreisender** [comment]
I am too late to review this. But the next time, please do not leave a bunch of those WIP commits behind. More expressive commit descriptions would, of course, be helpful. Maybe squash them, if they do not make sense on their own.

**Pull request merged**